src/Npgsql/NpgsqlConnection.cs

```diff
@@ -993,7 +993,25 @@ public int ProcessID
         /// <remarks>
         /// See https://www.postgresql.org/docs/current/static/sql-copy.html.
         /// </remarks>
-        public NpgsqlBinaryImporter BeginBinaryImport(string copyFromCommand)
+        public NpgsqlBinaryImporter BeginBinaryImport(string copyFromCommand) =>
+            BeginBinaryImport(copyFromCommand, async: false, CancellationToken.None).GetAwaiter().GetResult();
+
+        /// <summary>
+        /// Begins a binary COPY FROM STDIN operation, a high-performance data import mechanism to a PostgreSQL table.
+        /// </summary>
+        /// <param name="copyFromCommand">A COPY FROM STDIN SQL command</param>
+        /// <param name="cancellationToken">An optional token to cancel the asynchronous operation. The default value is None.</param>
+        /// <returns>A <see cref="NpgsqlBinaryImporter"/> which can be used to write rows and columns</returns>
+        /// <remarks>
+        /// See https://www.postgresql.org/docs/current/static/sql-copy.html.
+        /// </remarks>
+        public ValueTask<NpgsqlBinaryImporter> BeginBinaryImportAsync(string copyFromCommand, CancellationToken cancellationToken = default)
+        {
+            using (NoSynchronizationContextScope.Enter())
+                return BeginBinaryImport(copyFromCommand, async: true, cancellationToken);
+        }
+
+        async ValueTask<NpgsqlBinaryImporter> BeginBinaryImport(string copyFromCommand, bool async, CancellationToken cancellationToken = default)
```

Comment:
Shouldn't we pass the cancellation token to StartUserAction? Otherwise read operations don't get cancelled, right?

If so, same for the other operations.

Reply:
Not really, as in Init (and Read) for every COPY operation we call `StartNestedCancellableOperation`.

Reply:
Right, missed that (and need to remember how everything works after the vacation :sweat_smile:)